ARMmbed/mbedtls
# Oss-fuzz integration

## Description
Integration with oss-fuzz to do fuzz testing


## Status
IN DEVELOPMENT
but already consistent, useful
What is left to be done : 
- better corpuses (for client/server communication)
- use of different options for client/server communication

## Requires Backporting
No

## Migrations
No

## Additional comments
Found already one memory leak

## Todos


## Steps to test or reproduce
see https://github.com/google/oss-fuzz/blob/master/docs/new_project_guide.md
Basically :
- install docker
- Get my oss-fuzz pull request `git clone --depth 1 --branch mbedtls https://github.com/catenacyber/oss-fuzz.git oss-fuzz`
- Run the commands from oss-fuzz directory
  - `python infra/helper.py build_image mbedtls` 
  - `python infra/helper.py build_fuzzers --sanitizer address mbedtls` other sanitizers are available : memory and undefined
  - `python infra/helper.py run_fuzzer mbedtls fuzz_x509crl` or using another of the fuzz targets
- Wait for the fuzzer to find a bug
